pkg/sqlstore/store_user.go

```diff
@@ -17,6 +19,49 @@ type UserRows struct {
 	Rows    [][]interface{} `json:"rows"`
 }
 
+// UserValue wraps data from the db that may be raw json or any other value.
+type UserValue struct {
+	jsonValue  json.RawMessage
+	otherValue interface{}
+}
+
+// Value returns the underlying value.
+func (u *UserValue) Value() interface{} {
+	if u.jsonValue != nil {
+		return u.jsonValue
+	}
+	return u.otherValue
+}
+
+// Scan implements Scan.
+func (u *UserValue) Scan(src interface{}) error {
+	u.jsonValue = nil
+	u.otherValue = nil
+	switch src := src.(type) {
+	case string:
+		trimmed := strings.TrimLeft(src, " ")
+		if (strings.HasPrefix(trimmed, "{") || strings.HasPrefix(trimmed, "[")) && json.Valid([]byte(src)) {
+			u.jsonValue = []byte(src)
+		} else {
+			u.otherValue = src
+		}
+	case []byte:
+		tmp := src
+		u.otherValue = tmp
```

Comment:
Added this because of a dangerous sounding comment in the [`Scanner` declaration](https://pkg.go.dev/database/sql#Scanner) that 

> // Reference types such as []byte are only valid until the next call to Scan
> // and should not be retained. Their underlying memory is owned by the driver.
> // If retention is necessary, copy their values before the next call to Scan.

Reply:
Good call.

Important: `tmp := src` isn't the kind of copy `Scanner` is expecting. This is a shallow copy of the reference of the underlying slice, so `tmp` will still share the same bytes as `src`.

We should:
```
tmp := make([]byte, len(src))
copy(tmp, src)
```

Reply:
Done.